pkg/antctl/raw/check/util.go

```diff
@@ -103,3 +106,89 @@ func ExecInPod(ctx context.Context, client kubernetes.Interface, config *rest.Co
 	}
 	return stdout.String(), stderr.String(), nil
 }
+
+func NewDeployment(p DeploymentParameters) *appsv1.Deployment {
+	if p.Replicas == 0 {
+		p.Replicas = 1
+	}
+	replicas32 := int32(p.Replicas)
+	labels := map[string]string{
+		"name": p.Name,
+		"kind": p.Role,
+	}
+	var ports []corev1.ContainerPort
+	if p.Port > 0 {
+		ports = append(ports, corev1.ContainerPort{ContainerPort: int32(p.Port)})
+	}
+	var env []corev1.EnvVar
+	if p.Port > 0 {
+		env = append(env, corev1.EnvVar{Name: "PORT", Value: fmt.Sprintf("%d", p.Port)})
+	}
+	return &appsv1.Deployment{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:   p.Name,
+			Labels: labels,
+		},
+		Spec: appsv1.DeploymentSpec{
+			Replicas: &replicas32,
+			Selector: &metav1.LabelSelector{
+				MatchLabels: labels,
+			},
+			Template: corev1.PodTemplateSpec{
+				ObjectMeta: metav1.ObjectMeta{
+					Labels: labels,
+				},
+				Spec: corev1.PodSpec{
+					HostNetwork: p.HostNetwork,
+					Containers: []corev1.Container{
+						{
+							Name:            p.Name,
+							Image:           p.Image,
+							Ports:           ports,
+							Env:             env,
+							ImagePullPolicy: corev1.PullIfNotPresent,
+							Command:         p.Command,
+							VolumeMounts:    p.VolumeMounts,
+						},
+					},
+					Tolerations: p.Tolerations,
+					Volumes:     p.Volumes,
+					Affinity:    p.Affinity,
+				},
+			},
+		},
+	}
+}
+
+type DeploymentParameters struct {
+	Name         string
+	Role         string
+	Image        string
+	Replicas     int
+	Port         int
+	Command      []string
+	Affinity     *corev1.Affinity
+	Tolerations  []corev1.Toleration
+	Labels       map[string]string
+	VolumeMounts []corev1.VolumeMount
+	Volumes      []corev1.Volume
+	HostNetwork  bool
+}
+
+func WaitForDeploymentsReady(ctx context.Context, interval, timeout time.Duration, client kubernetes.Interface, antreaNamespace string, clusterName string, deployments ...string) error {
```

Comment:
The parameter `antreaNamespace` should be renamed to `namespace`. There is no reason for it to be called `antreaNamespace` and this function is also used by the `installation` command which creates Deployments in a "test Namespace", and not in the "antrea Namespace".